File: check_login.php
<!DOCTYPE html>
<html>
<head>
   <meta charset="utf-8">
   <title></title>
</head>
<body>
   <?php
   session_start();
   // $host = 'localhost';
   // $user = '';
   // $pw = '';
   // $db_name = '';
      // $mysqli = new mysqli($host, $user, $pw, $db_name); //db 연결
      
      //login.php에서 입력받은 id, password
      $username = $_POST['id'];
      $userpass = $_POST['pw'];
      
      // $q = "SELECT * FROM member WHERE id = '$username' AND pass = '$userpass'";
      // $result = $mysqli->query($q);
      // $row = $result->fetch_array(MYSQLI_ASSOC);
      
      //결과가 존재하면 세션 생성
      if ($username == 'webserv' && $userpass == 'no') {
         $_SESSION['username'] = $username;
         $_SESSION['name'] = 'Jacob';
         echo "<script>location.replace('index.php');</script>";
         exit;
      }
      else if ($username == 'irc' && $userpass == 'yes') {
         $_SESSION['username'] = $username;
         $_SESSION['name'] = 'Daniel';
         echo "<script>location.replace('index.php');</script>";
         exit;
      }
      else {
         echo "<script>alert('Invalid username or password')</script>";
         echo "<script>location.replace('login.php');</script>";
         exit;
      }
      ?>
   </body>

File: py-cgi/check_login.py
#!/usr/bin/env python3
import cgi
import os
import sys
from datetime import datetime, timedelta

def set_cookie(cookie_name, cookie_value, expires=False):
    expires_time = datetime.now() + timedelta(hours=1)
    cookie = "Set-cookie: " + cookie_name + "=" + cookie_value + ";"
    if expires:
        cookie += " Expires=" + expires_time.strftime('%a, %d %b %Y %H:%M:%S GMT')
    return cookie

def body_upper():
    body = "<!DOCTYPE html> <html> <head> <meta charset=""utf-8""> <title> </title> </head> <body>"
    print(body)


# Get the form data
form = cgi.FieldStorage()
username = form.getvalue("id")
password = form.getvalue("pw")

# Check if the username and password are correct
if username == "webserv" and password == "no":
    session_server = os.environ.get('PY_SESSION')
    if session_server:
        os.environ["PY_SESSION"] = session_server + ";" + username + "-Jacob=1"
    else:
        os.environ["PY_SESSION"] = username + "-Jacob=1"
    print(set_cookie("session_id", username + "-Jacob=1"))
    print("")
    body_upper()
    print("<script>location.replace('index.py');</script>")
    print("</body>")
elif username == "irc" and password == "yes":
    session_server = os.environ.get('PY_SESSION')
    if session_server:
        os.environ["PY_SESSION"] = session_server + ";" + username + "-David=2"
    else:
        os.environ["PY_SESSION"] = username + "-David=2"
    print(set_cookie("session_id", username + "-David=2"))
    print("")
    body_upper()
    print("<script>location.replace('index.py');</script>")
    print("</body>")
else:    
    body_upper()
    print("<script>alert('Invalid username or password')</script>")
    print("<script>location.replace('login.py');</script>")
    print("</body>")

